lib/spack/spack/test/conftest.py

```diff
@@ -683,3 +685,32 @@ def invalid_spec(request):
     """Specs that do not parse cleanly due to invalid formatting.
     """
     return request.param
+
+#######
+# Mock Configure executable
+#######
+
+
+@pytest.fixture(scope="function")
+def mock_configure():
+    """Create a mock configure in a directory"""
+    tmpdir = tempfile.mkdtemp()
+    configure_exe = os.path.join(tmpdir, "configure")
+    with open(configure_exe, "w") as file:
+        file.write("#!/bin/sh\n")
+        file.write("echo $@")
+    os.chmod(configure_exe, 0o700)
+
+    path_put_first("PATH", [tmpdir])
+
+    yield
+
+    shutil.rmtree(tmpdir)
```

Comment:
A couple things:

* You can use the tmpdir_factory fixture which will clean this up automatically (see `conftest.py`). That's not essential but it would follow the format of other tests
* While this should be sufficient, could you additionally restore the `PATH` environment variable? Something like `temp_env` in the `flag_handlers` test module would be excellent (you could move that to conftest and have `flag_handlers` and `mock_configure` use that).